packages/ui/src/machines/authenticator/actors/signIn.ts

```diff
@@ -104,6 +104,11 @@ export function signInActor({ services }: SignInMachineOptions) {
                     actions: ['setUser', 'setChallengeName'],
                     target: '#signInActor.setupTOTP',
                   },
+                  {
+                    cond: 'shouldSelectMfaType',
+                    actions: ['setUser', 'setChallengeName'],
+                    target: 'rejected',
+                  },
```

Comment:
I get a bit lost in the state machine, can you explain what is happening here?

Reply:
Sure. `cond` to check the user `shouldSelectMfaType` to be true, then `rejected`. 

The actions are copied from `shouldSetupTOTP` above because we'll only need to change `rejected` to something like `#signInActor.setupMfaType` once we add the feature.

Reply:
Couple follow up questions?
- in what scenarios is the condition reached?
- Are the actions what should happen for this condition of changing `rejected` in the future?

Reply:
> in what scenarios is the condition reached?

The scenario is in the description or https://github.com/aws-amplify/amplify-ui/issues/3767

The Admin add options from Amplify Console to allow user to choose MFA type, so Amplify UI would get a `SELECT_MFA_TYPE`.

> Are the actions what should happen for this condition or for ease of changing rejected in the future?

the actions are for ease of changing in the future.

Reply:
I don't think we should be factoring in the future changes, let's keep our changes focused on the fix

Reply:
FYI, this condition can be reached without admin doing this directly on AWS console: https://github.com/aws-amplify/amplify-js/issues/4646#issuecomment-1462908573